### PR TITLE
[add] support extra dev requirements

### PR DESCRIPTION
**Pull request by simahawk**

### CODE REVIEW

File: 9.0/Dockerfile [hunk 1]
```diff
@@ -2,11 +2,10 @@ FROM debian:jessie
 MAINTAINER Camptocamp
 
 # create the working directory and a place to set the logs (if wanted)
-RUN mkdir -p /opt/odoo /var/log/odoo
+RUN mkdir -p /opt/odoo/requirements /var/log/odoo
 
 WORKDIR "/opt/odoo"
-
-COPY ./base_requirements.txt ./
+COPY ./requirements/*requirements.txt ./requirements/
```
Comment (guewen) [inline 9.0/Dockerfile]: `COPY ./requirements/ ./requirements/`

File: 9.0/Dockerfile [hunk 2]
```diff
@@ -2,11 +2,10 @@ FROM debian:jessie
 MAINTAINER Camptocamp
 
 # create the working directory and a place to set the logs (if wanted)
-RUN mkdir -p /opt/odoo /var/log/odoo
+RUN mkdir -p /opt/odoo/requirements /var/log/odoo
```
Comment (guewen) [inline 9.0/Dockerfile]: change not needed if we copy the directory as I propose below

File: 9.0/Dockerfile [hunk 3]
```diff
@@ -104,4 +102,4 @@ ONBUILD COPY ./setup.py /opt/odoo/
 ONBUILD COPY ./VERSION /opt/odoo/
 ONBUILD COPY ./migration.yml /opt/odoo/
 # need to be called at the end, because it installs . and src
-ONBUILD RUN cd /opt/odoo && pip install -r src_requirements.txt
+ONBUILD RUN cd /opt/odoo && find . -maxdepth 1 -name "*requirements.txt" -exec pip install -r {} \;
```
Comment (guewen) [inline 9.0/Dockerfile]: It won't install `src_requirements.txt` isn't it? Shouldn't it be:

```
ONBUILD RUN cd /opt/odoo && pip install -r requirements/src_requirements.txt && find . -maxdepth 1 -name "*requirements.txt" -exec pip install -r {} \;
```

Comment (simahawk) [inline 9.0/Dockerfile]: oops, right :)

File: 9.0/Dockerfile (hunk 1 above)
Comment (simahawk) [inline 9.0/Dockerfile]: +1

**Comment by simahawk** [comment]
as we agreed: I pushed this to [dev_requirements](https://github.com/camptocamp/docker-odoo-project/tree/dev_requirements) branch to produce a testable image.

**Comment by simahawk** [comment]
REMINDER: if this works update the example project too and its readme

**Comment by guewen** [comment]
I don't remember what was the status on this, does it work?

**Comment by simahawk** [comment]
my test failed. Sounds like extra requirements are not there when they should ONBUILD.
Unless it works for you I'd say we should drop this and keep it only in the project setup which works for sure.

**Comment by guewen** [comment]
> Unless it works for you I'd say we should drop this and keep it only in the project setup which works for sure.

:+1: Thanks for trying :)

**Pull request closed without merging**